packages/medusa/src/models/order-edit.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+} from "typeorm"
+
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+import { resolveDbType } from "../utils/db-aware-column"
+import { OrderItemChange } from "./order-item-change"
+import { SoftDeletableEntity } from "../interfaces"
+import { generateEntityId } from "../utils"
+import { LineItem } from "./line-item"
+import { Order } from "./order"
+import { User } from "./user"
+
+@FeatureFlagEntity("order_editing")
+export class OrderEdit extends SoftDeletableEntity {
+  @Column({ nullable: true })
+  order_id: string
+
+  @ManyToOne(() => Order, (o) => o.edits)
+  @JoinColumn({ name: "order_id" })
+  order: Order
+
+  @OneToMany(() => OrderItemChange, (oic) => oic.order_edit, {
+    eager: true,
+    cascade: true,
+  })
+  changes: OrderItemChange[]
+
+  @Column({ nullable: true })
+  internal_note?: string
+
+  @Column({ nullable: true })
+  created_by: string
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "created_by" })
+  created_by_user: User
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  created_at: Date
+
+  @Column({ nullable: true })
+  requested_by?: string
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "requested_by" })
+  requested_by_user: User
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz"), nullable: true })
+  requested_at?: Date
+
+  @Column({ nullable: true })
+  confirmed_by?: string // customer or user ID
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz"), nullable: true })
+  confirmed_at?: Date
+
+  @Column({ nullable: true })
+  declined_by?: string // customer or user ID
+
+  @Column({ nullable: true })
+  declined_reason?: string
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz"), nullable: true })
+  declined_at?: Date
+
+  @Column({ nullable: true })
+  canceled_by?: string
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "canceled_by" })
+  canceled_by_user: User
```

Comment:
suggestion: same as above (remove FK)

Reply:
should we then remove `declined_by` since `canceled_by: [customer_id]` should be the same thing?

Reply:
I am not sure they describe the same intentions